src/serializer/ResidualHeapSerializer.js

```diff
@@ -1154,13 +1154,28 @@ export class ResidualHeapSerializer {
 
   _getPropertyAssignmentStatement(
     location: BabelNodeLVal,
-    value: BabelNodeExpression,
+    value: Value,
     mightHaveBeenDeleted: boolean,
     deleteIfMightHaveBeenDeleted: boolean = false
   ): BabelNodeStatement {
-    let assignment = t.expressionStatement(t.assignmentExpression("=", location, value));
     if (mightHaveBeenDeleted) {
-      let condition = t.binaryExpression("!==", value, this._serializeEmptyValue());
+      let serializedValue = this.serializeValue(value);
```

Comment:
Pull this definition out, replace call to `this.serializeValue(value)` in else branch with `serializedValue`.